### PR TITLE
Adds a modal component

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
### Issue:
- #911
- https://github.com/Real-Dev-Squad/website-status/issues/920

### Description:
Adds a Modal component and its tests.

### Dev Tested:
- [x] Yes

### Images/video of the change:

https://github.com/Real-Dev-Squad/website-status/assets/98796547/39392319-1c74-4add-981f-7a482f0ea916

### Coverage: 
Screenshot 2023-10-17 at 8 23 18 AM

### CODE REVIEW

**Review by iamitprakash — APPROVED** [review]

**Review by lakshayman — APPROVED** [review]

**Comment by ajoykumardas12** [comment]
Is the empty modal file intended?
Rest looks good.

**Comment by ajoykumardas12** [comment]
> > Is the empty modal file intended? Rest looks good.
> 
> Yes. Check #958

I was talking about this file: `src/components/Modal/Modal`
Don't see anything in this file in #958 

Screenshot

File: src/components/Modal/Modal
[the remark this answers is not in the document]
Comment (Ajeyakrishna-k) [reply src/components/Modal/Modal:1]: Removed it in PR : #958 

[Click here](https://github.com/Real-Dev-Squad/website-status/pull/958/commits/15d8b5ddeccbb25d6cb579d0f2d368f01844cc66) to view the changes.

**Comment by Ajeyakrishna-k** [comment]
> > > Is the empty modal file intended? Rest looks good.
> > 
> > 
> > Yes. Check #958
> 
> I was talking about this file: `src/components/Modal/Modal` Don't see anything in this file in #958
> 
> Screenshot

Removed.

**Review by ajoykumardas12 — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**